lib/host/gen_helpers.js

```diff
@@ -211,37 +211,31 @@ const genHelpers = (state = {}, adapter) => {
     },
 
     /**
-     * Submits an order after a delay, and adds it to the active order set on
+     * Submits an order and adds it to the active order set on
      * the AO state. Emits errors if the order fails to submit; retries up to
      * MAX_SUBMIT_ATTEMPTS in the case of balance evaluation errors.
      *
      * @param {object} state - current AO instance state
-     * @param {number} delay - delay in milliseconds
      * @param {object} order - order model
      * @param {number} attempt - attempt count, max is 10 (set as a constant)
      * @returns {object} nextState
      * @example
-     * await submitOrderWithDelay(state, 100, order)
+     * await submitOrder(state, order)
      */
-    submitOrderWithDelay: async (state = {}, delay, order, attempt = 1) => {
+    submitOrder: async (state = {}, order, attempt = 1) => {
       logHelperCall(
-        'submitOrderWithDelay (%dms, attempt %d): %s',
-        delay, attempt, order.toString()
+        'submitOrder (attempt %d): %s',
+        attempt, order.toString()
       )
 
       if (attempt > MAX_SUBMIT_ATTEMPTS) { // note state assumed already patched
+        // TODO cancel/delete from state an order defined in /submit_all_orders.js
         return state
       }
 
       const { connection } = state
-      const orderPatch = {
-        [order.cid + '']: order
-      }
 
-      // Note that we don't wait for the order to submit here, since it might
-      // fill immediately triggering a fill event before we patch the state
-      // orders/allOrders objects
-      adapter.submitOrderWithDelay(connection, delay, order).then((order) => { // eslint-disable-line
```

Comment:
did you check this note?

```
      // Note that we don't wait for the order to submit here, since it might
      // fill immediately triggering a fill event before we patch the state
      // orders/allOrders objects
```

Reply:
just checking, this is related to this LOC in submit_all_orders.js:

```js
nextState = await submitOrderWithDelay(nextState, delay, o)
```

Reply:
covered by the internal `withAOUpdate` now, so we are good

Reply:
Of course. This is no longer required since the loop has been replaced with Promise.all 

https://github.com/bitfinexcom/bfx-hf-algo/blob/a0b4bceacf9bbe34605e1c28e4098826b41e5008/lib/host/events/submit_all_orders.js#L51-L58